userspace/libsinsp/sinsp.cpp

```diff
@@ -542,6 +544,44 @@ void sinsp::fill_tp_of_interest(scap_open_args *oargs, const std::unordered_set<
 	}
 }
 
+/*=============================== INTERESTING TRACEPOINTS/SYSCALLS ===============================*/
+
+/*=============================== BUFFER DIMENSION VALIDATION ===============================*/
+
+static void validate_driver_buffer_num_pages(uint64_t driver_buffer_num_pages)
+{
+	std::unordered_set<uint32_t> allowed_dimensions;
+	
+	/* Allowed page numbers are all the power of 2 from 128(2^7) pages to 256Kpages(2^18) */
+	for(int i = 7; i <= 18; i++)
+	{
+		uint32_t val = 1<<i;
+		allowed_dimensions.insert(val);
+	}
```

Comment:
Why do we compute this at runtime? Can't we just have the values specified in the constructor of `allowed_dimensions`?

Reply:
Maybe using `constexpr` would make it so it is computed at compile time, but I'm not that well versed in its use 😅

Reply:
I'd also move the validation as near as possible to the drivers, ie: in libscap.

Reply:
Make the validation logic a function in ringbuffer/devset.c called from each engine that needs it? That there even _is_ a ringbuffer is an engine's implementation detail

Reply:
It could be an idea, but it is always a check in userspace :thinking: the point that I don't understand here is how this value can change from the one received by `sinsp` since we have no other methods to set it :thinking: A kernel side check probably is not essential but has more sense, WDYT?

Reply:
> I don't understand here is how this value can change from the one received by sinsp

```
echo 3 > /sys/module/scap/per_cpu_buffer_dim
```

or

```
bc -l <<< '2 ^ 63' > /sys/module/scap/per_cpu_buffer_dim
```

Look ma, no sinsp!

A kernel side check is absolutely essential. The kernel/user boundary here is just like the client/server one in web apps: you do client (user) side validation for UX (e.g. faster or better error messages) but the server (kernel) side validation is the one that matters.